tests/unit/test_connection.py

```diff
@@ -42,6 +42,12 @@ def setUp(self):
             'Accept': 'application/json',
             'Content-Type': 'application/json'
         }
+        self.default_headers_with_etag_validation_off = {
+            'X-API-Version': 200,
```

Comment:
Good catch!
I believe the default API Version should be changed to 300.
Since it's not related with the implemented feature, we can do this change later.